### ISSUE TITLE
Render and Main process does not restart when the files are changed

### ISSUE THREAD

**Issue by cawa-93** [SimulatedGREG/electron-vue#319]
##### Describe the issue / bug.
I do not understand why, but when you change the files in the `/src` directory, the build is not updated. I'll completely restart the application

##### Tell me about your development environment.
  * Node version: v8.1.4
  * NPM version: 5.3.0
  * vue-cli version: 2.8.2
  * Operating System: antergos (Arch Linux)

**Comment by develar** [comment]
In my case works (I have common code outside of renderer/main directories). I suggest you to use https://github.com/electron-userland/electron-webpack (#310)  (example project — https://github.com/electron-userland/electrify).

electron-webpack supports HMR for both renderer and main process, without reloading (to avoid restarts for main process, you need to implement https://webpack.js.org/api/hot-module-replacement/).

**Comment by develar** [comment]
https://medium.com/@develar/electron-very-fast-developer-workflow-with-webpack-hmr-e2a2e23590ad

**Comment by cawa-93** [comment]
You can try it
```
$ git clone git@github.com:cawa-93/command-editor.git
$ cd command-editor
$ git checkout update
$ npm i
$ npm run dev
```
The compiler does not respond to file changes. And I do not understand why.


My Dependency List:
```
├── axios@0.16.2
├── babel-core@6.25.0
├── babel-eslint@7.2.3
├── babel-loader@7.1.1
├── babel-plugin-transform-runtime@6.23.0
├── babel-preset-env@1.6.0
├── babel-preset-stage-0@6.24.1
├── babel-register@6.24.1
├── babili-webpack-plugin@0.1.2
├── cfonts@1.1.3
├── chalk@1.1.3
├── copy-webpack-plugin@4.0.1
├── cross-env@5.0.1
├── css-loader@0.28.4
├── del@2.2.2
├── devtron@1.4.0
├── electron@1.7.5
├── electron-builder@19.16.1
├── electron-debug@1.2.0
├── electron-devtools-installer@2.2.0
├── eslint@3.19.0
├── eslint-config-standard@10.2.1
├── eslint-friendly-formatter@3.0.0
├── eslint-loader@1.9.0
├── eslint-plugin-html@2.0.3
├── eslint-plugin-import@2.7.0
├── eslint-plugin-node@4.2.3
├── eslint-plugin-promise@3.5.0
├── eslint-plugin-standard@3.0.1
├── extract-text-webpack-plugin@2.1.2
├── fast-csv@2.4.0
├── file-loader@0.11.2
├── html-webpack-plugin@2.29.0
├── iconv-lite@0.4.18
├── jschardet@1.5.0
├── json-loader@0.5.4
├── lodash.clonedeep@4.5.0
├── material-design-icons@3.0.1
├── multispinner@0.2.1
├── style-loader@0.18.2
├── transliteration@1.6.2
├── url-loader@0.5.9
├── vue@2.4.1
├── vue-electron@1.0.6
├── vue-html-loader@1.2.4
├── vue-loader@12.2.2
├── vue-router@2.7.0
├── vue-style-loader@3.0.1
├── vue-template-compiler@2.4.1
├── vuetify@0.14.5
├── vuex@2.3.1
├── webpack@2.7.0
├── webpack-dev-server@2.5.1
└── webpack-hot-middleware@2.18.2
```

**Comment by develar** [comment]
@cawa-93 Don't get me wrong, but could you please try to migrate to electron-webpack (in any case will be merged soon, #300)?

**Comment by develar** [comment]
@cawa-93 I see only `index.ejs` file in the src, all other files in the main/renderer. Or do you mean that nothing is updated if you change any file in the src and child dirs?

**Comment by cawa-93** [comment]
Yes. I had to enter the vails in the directories `main` and `render`

**Comment by cawa-93** [comment]
For example. I started the command `npm run dev`. The application opened. I changed the text of the button (suppose it's in the file `/src/render/component/btn.vue`). I expected the Renderer Process to be automatically recompiled. But this does not happen.

**Comment by develar** [comment]
electron-webpack supports HMR — please migrate to it and try :)

Migration is easy —
1. [Install](https://github.com/electron-userland/electron-webpack#install), 
2. Change [package scripts](https://github.com/electron-userland/electron-webpack#package-scripts). 
3. Remove `.electron-vue` directory.
4. If you use electron-builder, make sure that you use version 19.16.1 +

You can see sample project — http://github.com/electron-userland/electrify

**Comment by cawa-93** [comment]
I get this error:
```
Error: Cannot find entry file src/renderer/index.ts
```
Is it by default designed for a project on a typescript?

**Comment by develar** [comment]
I will check it now. Сейчас поправлю, если удобнее, можно в слаке напрямую (http://slackin.electron.build) писать.

**Comment by develar** [comment]
@cawa-93 Fixed in 0.5.2, thanks for report.

**Comment by SimulatedGREG** [comment]
@cawa-93 

I'm not sure what issue you are running into, as I am not able to reproduce this either. Are you able to test this on another system?

![kapture 2017-07-20 at 3 31 55](https://user-images.githubusercontent.com/8327811/28408119-06527c9c-6cfc-11e7-80d9-49aae21963a6.gif)

**Comment by cawa-93** [comment]
@SimulatedGREG try `update` branche

**Comment by develar** [comment]
@cawa-93 Forgot to mention — electron-webpack requires webpack 3.0.

**Comment by cawa-93** [comment]
Right. How did @develar  advise me, I installed the electron-webpack
@0.5.2 and webpack@3.3.0 However, the result remained the same: the collector does not respond to file changes. See [this](https://drive.google.com/file/d/0BzfqYvbayZQpZkthRUFMWHN5OEE/view?usp=sharing)

**Comment by SimulatedGREG** [comment]
@cawa-93 

I think this might be an OS level issue with Arch Linux. There is a slim chance you are reaching the `inotify` user watch limit. Basically saying that when your text editor and `webpack` are watching all your source, assets, and npm files you are reaching a limit the OS let's you catch on file changes. Take a look at these links about how to increase that limit and see if that helps.

This link is part of a separate ruby package called `listen`, but the documentation here is still related to OS-level filesystem watching.
https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers

Related issue with `chokidar` that `webpack` uses internally.
https://github.com/paulmillr/chokidar/issues/294

**Comment by cawa-93** [comment]
It seems that it is so. After restarting the laptop everything started to work as expected (although I do not know how long). @develar, @SimulatedGREG, Thanks for the help.

**Comment by develar** [comment]
@cawa-93 have you seen my patch (slack)?

> reaching a limit the OS let's you catch on file changes

To fix this it, electron-webpack 0.6.0 watches only source directory (`src/**/*`, but source directory for other process also excluded). https://github.com/electron-userland/electron-webpack/blob/master/packages/electron-webpack/src/plugins/WatchMatchPlugin.ts

**Comment by jhonyjss** [comment]
For me to work Main and Render reload I Just put

```javascript
// main.js and index.js
if (module.hot) { module.hot.accept() }

// dev-runner.js

const server = new WebpackDevServer(
      compiler,
      {
        contentBase: path.join(__dirname, '../'),
        quiet: true,
        hot: true,
        liveReload: true,
        watchOptions: {
          poll: true
        },
```
